src/weak_memory.rs

```diff
@@ -0,0 +1,271 @@
+use std::{
+    cell::{Ref, RefCell, RefMut},
+    collections::VecDeque,
+};
+
+use rustc_const_eval::interpret::{AllocRange, InterpResult, ScalarMaybeUninit};
+use rustc_data_structures::fx::FxHashMap;
+use rustc_target::abi::Size;
+
+use crate::{
+    data_race::{GlobalState, ThreadClockSet},
+    RangeMap, Tag, VClock, VTimestamp, VectorIdx,
+};
+pub type AllocExtra = StoreBufferAlloc;
+#[derive(Debug, Clone)]
+pub struct StoreBufferAlloc {
+    /// Store buffer of each atomic object in this allocation
+    store_buffer: RefCell<RangeMap<StoreBuffer>>,
```

Comment:
Why does this need to be a RefCell? Does a load affect the store buffer?

Hm yeah I guess it does. Still worth a comment.